prometheus/metric.go

```diff
@@ -187,6 +187,15 @@ func (m *withExemplarsMetric) Write(pb *dto.Metric) error {
 		pb.Counter.Exemplar = m.exemplars[len(m.exemplars)-1]
 	case pb.Histogram != nil:
 		for _, e := range m.exemplars {
+			if pb.Histogram.Schema != nil {
+				if *pb.Histogram.Schema > math.MinInt32 && e.GetTimestamp() != nil {
```

Comment:
While we use `math.MinInt32` as a schema within the `histogram` struct to mark the absence of sparse buckets, we do not use the same marker in the protobuf. In the protobuf, we know it represents a native histogram if the zero threshold or the zero count or the number of negative spans or the number of positive spans are not zero. (If need be, we add a no-op span.)

So what we have to check is something like the following:
```Go
h := pb.Histogram
if (*h.ZeroThreshold != 0 || *h.ZeroCount != 0 || len(h.PositiveSpan) != 0 || len(h.NegativeSpan) != 0) && e.GetTimestamp() != nil {
```